src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -67,16 +72,104 @@ internal struct ParsedStat
             //internal long cguest_time;
         }
 
+        internal struct ParsedMapsModule
+        {
+            internal string FileName;
+            internal KeyValuePair<long, long> AddressRange;
+        }
+
         internal static string GetStatFilePathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + StatFileName;
         }
 
+        internal static string GetMapsFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + MapsFileName;
+        }
+
         internal static string GetTaskDirectoryPathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + TaskDirectoryName;
         }
 
+        internal static IEnumerable<ParsedMapsModule> ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadAllLines(GetMapsFilePathForProcess(pid)));
```

Comment:
I'm curious, why `File.ReadAllLines` and not `File.ReadLines`? `ReadAllLines` involves some extra allocations in its implementation, not to mention the array allocation. If there is a good reason to stick with `File.ReadAllLines`, perhaps `ParseMapsModulesCore`'s parameter should be changed from `IEnumerable<string>` to `string[]` to avoid the enumerator allocation when enumerating the lines.